src/pages/Compare/index.spec.tsx

```diff
@@ -886,6 +916,287 @@ describe(__filename, () => {
     expect(dispatch).not.toHaveBeenCalled();
   });
 
+  it('dispatches fetchDiff and fetchVersionFile for the next file with diff', () => {
+    const addonId = 10;
+    const baseVersionId = 44;
+    const headVersionId = 55;
+    const file1 = 'file1.json';
+    const file2 = 'file2.json';
+    const version = createExternalVersionWithEntries(
+      [
+        { path: file1, status: 'M' },
+        { path: 'fileWithoutChange', status: '' },
+        { path: file2, status: 'M' },
+      ],
+      { id: headVersionId },
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      { selected_file: file1, diff: fakeExternalDiff },
+    ) as ExternalVersionWithDiff;
```

Comment:
After my change suggestion on the helper, maybe you won't need this? I'm not sure. It's complaining because `file.diff` could potentially be `undefined`. Another solution (maybe) would be to define `createExternalVersionDiffWithEntries` that wraps the other helper.

Reply:
Type assertions are risky because they can hide mistakes. Sometimes we do need them, though.